vllm/model_executor/models/mlp_speculator.py

```diff
@@ -137,8 +180,10 @@ def generate_proposals(
     def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
         params_dict = dict(self.named_parameters())
         for name, loaded_weight in weights:
-            param = params_dict.get(name.replace("speculator.", ""))
-            if param is not None:
+            try:
+                param = params_dict[name.replace("speculator.", "")]
                 weight_loader = getattr(param, "weight_loader",
                                         default_weight_loader)
                 weight_loader(param, loaded_weight)
+            except:
+                pass
```

Comment:
Probably we can remove this try/except block now

Reply:
@tdoublep changed this dict lookup to use `.get()` because we don't load all of the layers in the case the the configured num lookahead tokens < num heads. I probably should have added a comment explaining that though.

Reply:
ah right yeah. sorry that got lost in my merge somehow.